daemon/fqdn.go

```diff
@@ -106,17 +107,42 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 		fqdn.StartDNSPoller(d.dnsPoller)
 	}
 
+	fqdn.FQDNMaxIPperHost = option.Config.ToFQDNsMaxIPsPerHost
+	// Controller to cleanup TTL expired entries from the DNS policies.
+	controller.NewManager().UpdateController("dns-garbage-collector-job", controller.ControllerParams{
+		RunInterval: 1 * time.Minute,
+		DoFunc: func() error {
+
+			namesToClean := []string{}
+			endpoints := endpointmanager.GetEndpoints()
+			for _, ep := range endpoints {
+				namesToClean = append(namesToClean, ep.DNSHistory.GC()...)
+			}
+
+			namesToClean = fqdn.KeepUniqueNames(namesToClean)
+			if len(namesToClean) == 0 {
+				return nil
+			}
+			// A second loop is needed to update the global cache from the
+			// endpoints cache
+			for i, ep := range endpoints {
```

Comment:
Also mention: The first update from an endpoint will clear the data in cfg.Cache for each of namesToClean. Looping this way is generally safe despite not locking; If a new lookup happens during these updates the new DNS data will be reinserted from the endpoint.DNSHistory cache that made the request.